cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java

```diff
@@ -266,6 +268,8 @@
       /** for the address (hostname) of the dataset server. */
       public static final String ADDRESS = "dataset.service.bind.address";
       public static final String PORT = "dataset.service.bind.port";
+      public static final String ANNOUNCE_ADDRESS = "dataset.service.announce.address";
```

Comment:
Do we actually need two different config? Shouldn't we have one for things that run inside the master process?